packages/suite-desktop-core/e2e/support/pageActions/marketActions.ts

```diff
@@ -0,0 +1,113 @@
+    readonly bestOfferAmount: Locator;
+    readonly buyBestOfferButton: Locator;
+    readonly youPayInput: Locator;
+    readonly buyOffersPage: Locator;
+    readonly compareButton: Locator;
+    readonly quotes: Locator;
+    readonly quoteOfProvider = (provider: string) =>
+        this.page.getByTestId(`@coinmarket/offers/quote-${provider}`);
+    readonly quoteProvider: Locator;
+    readonly quoteAmount: Locator;
+    readonly selectThisQuoteButton: Locator;
+    readonly modal: Locator;
+    readonly buyTermsConfirmButton: Locator;
+    readonly confirmOnTrezorButton: Locator;
+    readonly confirmOnDevicePrompt: Locator;
+    readonly tradeConfirmation: Locator;
+    readonly tradeConfirmationCryptoAmount: Locator;
+    readonly tradeConfirmationProvider: Locator;
+    readonly tradeConfirmationContinueButton: Locator;
+
+    constructor(private page: Page) {
+        this.offerSpinner = this.page.getByTestId('@coinmarket/offers/loading-spinner');
+        this.layout = this.page.getByTestId('@coinmarket');
+        this.form = this.page.getByTestId('@coinmarket/form');
+        this.bestOfferProvider = this.page.getByTestId('@coinmarket/offers/quote/provider');
+        this.bestOfferAmount = this.page.getByTestId('@coinmarket/form/offer/crypto-amount');
+        this.buyBestOfferButton = this.page.getByTestId('@coinmarket/form/buy-button');
+        this.youPayInput = this.page.getByTestId('@coinmarket/form/fiat-input');
+        this.buyOffersPage = this.page.getByTestId('@coinmarket/buy-offers');
+        this.compareButton = this.page.getByTestId('@coinmarket/form/compare-button');
+        this.quotes = this.page.getByTestId('@coinmarket/offers/quote');
+        this.quoteProvider = this.page.getByTestId('@coinmarket/offers/quote/provider');
+        this.quoteAmount = this.page.getByTestId('@coinmarket/offers/quote/crypto-amount');
+        this.selectThisQuoteButton = this.page.getByTestId(
+            '@coinmarket/offers/get-this-deal-button',
+        );
+        this.modal = this.page.getByTestId('@modal');
+        this.buyTermsConfirmButton = this.page.getByTestId(
+            '@coinmarket/buy/offers/buy-terms-confirm-button',
+        );
+        this.confirmOnTrezorButton = this.page.getByTestId(
+            '@coinmarket/offer/confirm-on-trezor-button',
+        );
+        this.confirmOnDevicePrompt = this.page.getByTestId('@prompts/confirm-on-device');
+        this.tradeConfirmation = this.page.getByTestId('@coinmarket/selected-offer');
+        this.tradeConfirmationCryptoAmount = this.page.getByTestId(
+            '@coinmarket/form/info/crypto-amount',
+        );
+        this.tradeConfirmationProvider = this.page.getByTestId('@coinmarket/form/info/provider');
+        this.tradeConfirmationContinueButton = this.page.getByTestId(
+            '@coinmarket/offer/continue-transaction-button',
+        );
+    }
+
+    openCoinMarket = async () => {
+        const walletPage = new WalletActions(this.page);
```

Comment:
I feel this is just a little bit cleaner then getting the fixture walletPage here thru constructor.
What do you think?
:thinking:

Reply:
Or should this method be completely in walletPage object and I should import that in test? Let me see locally

Reply:
Moved to WalletActions.